test/dotnet-msbuild.Tests/GivenDotnetStoreInvocation.cs

```diff
@@ -18,6 +18,11 @@ public class GivenDotnetStoreInvocation
         private static readonly string WorkingDirectory = 
             TestPathUtilities.FormatAbsolutePath(nameof(GivenDotnetStoreInvocation));
 
+        public GivenDotnetStoreInvocation()
+        {
+            Telemetry.Telemetry.CurrentSessionId = null;
```

Comment:
>Telemetry.Telemetry.CurrentSessionId = null; [](start = 12, length = 44)

Why?

Reply:
Exactly. Without this, some tests were failing randomly with the SessionId being set.

Reply:
OK, see other comment about how this is sprinkled around. Can we put it behind a method with an explanation.